lib/Basics/threads-posix.cpp

```diff
@@ -65,7 +65,7 @@ struct thread_data_t {
 ////////////////////////////////////////////////////////////////////////////////
 
 static void* ThreadStarter(void* data) {
-  arangodb::signals::maskAllSignals();
+  arangodb::signals::maskAllSignalsServer();
```

Comment:
I dislike calling the server-side variant here unconditionally, as the code in `lib` is used by the client-tools as well.
I have no doubt it will work for _now_, but it doesn't look very fool-proof.
We should **at least** leave a code comment here why we are unconditionally calling the server variant here. 
I would even prefer to have 2 different implementations for `maskAllSignals()` which are compiled separately in different TUs (one only for arangod, the other one for client-tools). That sounds like the cleanest solution to me.